rc/tools/doc.kak

```diff
@@ -165,3 +165,100 @@ define-command -params 1..2 \
 }
 
 alias global help doc
+
+define-command -params 1.. -docstring %{
+        doc-search <pattern> [topic...]: search for a pattern in the documentation
+        Display a menu listing all the search matches in the given topics, or the whole documentation if unspecified
+    } \
+    -shell-script-candidates %{
+        case "${kak_token_to_complete}" in
+            0) ;;
+            *) find "${kak_runtime}/doc/" -type f -name '*\.asciidoc' | sed 's,.*/,,; s,\.[^/]*$,,';;
+        esac
+    } doc-search %{ evaluate-commands %sh{
+        readonly SELECTIONS_FILE=$(mktemp "${TMPDIR:-/tmp}"/kak-doc-search.XXXXXXXX)
+        readonly SEARCH_PATTERN="${1}"
+
+        shift
+        if [ $# -eq 0 ]; then
+            eval set -- $(ls "${kak_runtime}/doc"/*.asciidoc | sed 's,.*/,,; s,\.[^/]*$,,')
+        else
+            for topic; do
+                if [ ! -e "${kak_runtime}/doc/${topic}.asciidoc" ]; then
+                    printf 'fail No such topic %%{%s}' "${topic}"
+                    exit 1
+                fi
+            done
+        fi
+
+        # NOTE: don't rely on the PATH to make sure the `kak` instance we run implements doc-searching
+        env KAK_DOC_SEARCH_TMP="${SELECTIONS_FILE}" \
+            KAK_DOC_SEARCH="${SEARCH_PATTERN}" \
+            "${kak_runtime}"/../../bin/kak -ui dummy -e "doc-search-impl $*"
```

Comment:
I might be able to optimize startup time here by adding `-n -E 'source …/doc.kak'`. Highlighters in the user configuration wouldn't make previously saved selection descriptions invalid, would they?